packages/app-check/src/internal-api.test.ts

```diff
@@ -284,6 +284,67 @@ describe('internal api', () => {
       );
       expect(token).to.deep.equal({ token: fakeRecaptchaAppCheckToken.token });
     });
+
+    it('reads any memory-cached debug token if in debug mode', async () => {
+      const clock = useFakeTimers();
```

Comment:
Are faketimers needed in these tests?

Reply:
I went down quite a rabbit hole of fixing flaky tests and I think hopefully I've gotten it? This is good because app check tests have been flaky for a while and I think it's important to try to get to the bottom of it.

- I removed unnecessary fake timers that only turned out to be needed because it made isValid() work with the really low `expireTimeMillis` I had (123). I set all expireTimeMillis to Date.now() + 60000.
- There were hanging promises caused by grecaptcha.ready() that caused the test suite to end early sometimes because I think karma interpreted resolved promises from a previous test as a done() of the currently running async tests. I stubbed grecaptcha when we weren't testing for it specifically
- Writing to real storage could cause inconsistent states between tests, stubbed the reads and writes by default